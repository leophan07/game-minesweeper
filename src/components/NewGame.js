import styled from 'styled-components';
import Button from './Button';

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  width: 200px;
  margin: 0 auto;
`;

export const PlayButton = styled(Button)`
  margin-top: 20px;
`;
