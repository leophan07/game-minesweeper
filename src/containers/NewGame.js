import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Content, PlayButton } from '../components/NewGame';

function NewGameContainer({ onPlayGame, isStarting }) {
  const [level, setLevel] = useState(9);

  return (
    <Content>
      <select onChange={(e) => setLevel(parseInt(e.target.value))}>
        <option value={9}>Beginner</option>
        <option value={16}>Advantage</option>
      </select>
      <PlayButton onClick={() => onPlayGame(level)}>
        {isStarting ? 'Please wait' : 'Play game'}
      </PlayButton>
    </Content>
  )
}

NewGameContainer.propTypes = {
  isStarting: PropTypes.bool.isRequired,
  onPlayGame: PropTypes.func.isRequired,
}

const mapStateToProps = (state) => {
  return {
    isStarting: state.isSetting,
  };
}

export default connect(mapStateToProps)(NewGameContainer);