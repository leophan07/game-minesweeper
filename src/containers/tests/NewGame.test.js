import React from 'react';
import NewGame from '../NewGame';
import renderWithRedux from '../../helpers/renderTest';

describe('<NewGameContainer />', () => {

  const playGameFn = jest.fn();

  it('should render and match the snapshot', () => {
    const { container } = renderWithRedux(
      <NewGame onPlayGame={playGameFn} isStarting={false} />
    );
    expect(container).toMatchSnapshot();
  });

  it ('should render text PleaseWait when change isSetting', () => {
    const { getByTestId } = renderWithRedux(<NewGame onPlayGame={playGameFn} />, {
      initialState: {
        isSetting: true,
      },
    });
    expect(getByTestId("button-play-game")).toHaveTextContent("Please wait");
  });
});