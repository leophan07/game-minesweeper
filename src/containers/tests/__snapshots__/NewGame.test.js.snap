// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewGameContainer /> should render and match the snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH dMjuZz"
  >
    <select>
      <option
        value="9"
      >
        Beginner
      </option>
      <option
        value="16"
      >
        Advantage
      </option>
    </select>
    <button
      class="sc-bdVaJa sc-htpNat gYrVGL"
      data-testid="button-play-game"
    >
      Play game
    </button>
  </div>
</div>
`;
